proxy/handler/request_handler.go

```diff
@@ -165,6 +168,30 @@ func (h *RequestHandler) HandleWebSocketRequest(iter route.EndpointIterator) {
 	h.reporter.CaptureWebSocketUpdate()
 }
 
+func (h *RequestHandler) SanitizeRequestConnection() {
```

Comment:
this 3x nested for-loop will run on every request, regardless of the `HopByHopHeadersToFilter` config setting atm.
can we make sure it runs only when there are actual values in the config property? don't want to waste time on a critical path.

something like
```suggestion
func (h *RequestHandler) SanitizeRequestConnection() {
if len(h.hopByHopHeadersToFilter) == 0 {
  return
}
```